### PR TITLE
VP-3939: Price facet filter problems

### PR DESCRIPTION
**Pull request by pushnitsa**
### Problem
When surf to categories, facet filtering is not working

### Solution
Fix problem

### Proposed of changes
Add term builder and other fixes

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

File: src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs [hunk 1]
```diff
@@ -106,22 +121,29 @@ public IRequestBuilder AddTerms(IEnumerable<string> terms)
...
+        }
+
         private void ParseFilters(string filterPhrase)
         {
+            var filters = new List<IFilter>();
+
             if (filterPhrase == null)
             {
                 return;
             }
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs:144]: Can you move filters initialization after guardian?

**Review by Vectorfield4 — APPROVED** [review]

**Review by yecli — COMMENTED** [review]
Please remove escaping if it is not needed space anymore:
https://github.com/VirtoCommerce/vc-module-experience-api/commit/cfda698e8ff495655b30066a9e0c8bf83eb28935

File: src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs [hunk 2]
```diff
@@ -76,16 +77,30 @@ public virtual IRequestBuilder FromQuery(IGetDocumentsByIdsQuery query)
...
             {
+                var aggregationFilterFieldName = (aggr.Filter as INamedFilter)?.FieldName;
+
                 var clonedFilter = SearchRequest.Filter.Clone() as AndFilter;
-                clonedFilter.ChildFilters = clonedFilter.ChildFilters
-                    .Where(x => !(x is INamedFilter namedFilter) || !namedFilter.FieldName.EqualsInvariant(aggr.FieldName))
-                    .ToList();
 
-                aggr.Filter = clonedFilter;
+                clonedFilter.ChildFilters = clonedFilter
```
Comment (tatarincev) [inline src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs:87]: Please add a small comment on what you do by this code

File: src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Mapping/FacetMappingProfile.cs [hunk 3]
```diff
@@ -1,58 +1,53 @@
...
-            CreateMap<TermAggregationRequest, TermFacetResult>().ConvertUsing((request, facet, context) =>
+            CreateMap<Aggregation, FacetResult>().ConvertUsing((request, facet, context) =>
             {
-                var aggregations = context.Items["aggregations"] as IList<AggregationResponse>;
-                var aggregation = aggregations.FirstOrDefault(x => x.Id.EqualsInvariant(request.Id) || x.Id.EqualsInvariant(request.FieldName));
-                if (aggregation != null)
+                FacetResult result = request.AggregationType switch
                 {
-                    return new TermFacetResult
+                    "attr" => new TermFacetResult
```
Comment (tatarincev) [inline src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Mapping/FacetMappingProfile.cs:19]: I suppose you forgot for true `RangeFacetResult` types in response.
Try to test by using these arguments ` filter:"price:(TO 100)"` with this include `range_facet { name }`

File: src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Queries/SearchProductQueryHandler.cs [hunk 4]
```diff
@@ -67,26 +78,39 @@ ISearchProvider searchProvider
...
             var expProducts = await ConvertIndexDocsToProductsWithLoadDependenciesAsync(searchResult.Documents, request);
 
-            return new SearchProductResponse
+            var criteria = new ProductIndexedSearchCriteria
+            {
+                StoreId = request.StoreId,
+                Currency = request.CurrencyCode,
+            };
+
+            var aggregations = await _aggregationConverter.ConvertAggregationsAsync(searchResult.Aggregations, criteria);
```
Comment (tatarincev) [inline src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Queries/SearchProductQueryHandler.cs:104]: Please add a small comment on what you do by this code

File: src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs (hunk 1 above)
Comment (pushnitsa) [reply src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs:145]: Done

**Pull request merged**